src/main/java/org/alliancegenome/curation_api/jobs/executors/GeneExecutor.java

```diff
@@ -57,18 +58,45 @@ public void runLoad(BulkLoadFile bulkLoadFile) {
 			String speciesName = manual.getDataType().getSpeciesName();
 			String dataType = manual.getDataType().name();
 
-			if (genes == null)
-				genes = new ArrayList<>();
+			List<String> geneCuriesLoaded = new ArrayList<>();
+			List<String> geneCuriesBefore = geneService.getCuriesBySpeciesName(speciesName);
+			
+			log.debug("runLoad: Before: total " + geneCuriesBefore.size());
+			
+			if (genes != null) {
+				bulkLoadFile.setRecordCount(genes.size() + bulkLoadFile.getRecordCount());
+				bulkLoadFileDAO.merge(bulkLoadFile);
+
+				APIResponse resp = runLoad(speciesName, genes, dataType, geneCuriesLoaded);
+				trackHistory(resp, bulkLoadFile);
+			}
```

Comment:
@markquintontulloch is this the correct logic?

Reply:
I think that would work, although the mechanism I discussed with Chris set the record count as 0 and reported that on the data loads page.  That would then also be consistent with the behaviour of the disease annotation executors.  So here you would have
```
if (genes == null)
    genes = new ArrayList<>();
```
and then all the other code would run regardless.

Reply:
Talking of disease annotation executors, doesn't this cleanup tracking also need to be applied to disease annotations?

Reply:
Yes I would guess, on the annotations.

Reply:
So then I would propose that we track the deleted records on the BulkLoadFileHistory. We get a blank file we delete everything and put a field on there to show it. If that part of the load "fails" then the load fails.

Reply:
I guess that would be a nice addition. My main concern here though is that we are not undoing what has been implemented already without a different solution in place.

I had thought this PR was supposed to be a complete solution to SCRUM-2455, hence my comment about applying to disease annotations.  I see now though that the reporting part is yet to be addressed though.

Reply:
Cool - so presumably for empty ingest sets you would have something like "Records completed - 0; Records failed - 0; Records deleted - 100".  Keeping the logic of running the load with an empty list seems like an easy way of achieving the records completed/failed part, at least for now.

Reply:
Ya... and I am also adding a Records Failed Deleting also which will capture the exception vs print to the logs and store it as an exception like the rest. Then once we get the exception download working you should have all the info to fix the file, or in our case fix the foreign key issues. (Feature) I think it would also be nice to have "category's" of exceptions and them be grouped. So you could see the different kinds of errors instead of having to scroll through everything.. but anyway.